crates/turborepo-lib/src/run/mod.rs

```diff
@@ -222,17 +225,31 @@ impl<'a> Run<'a> {
             }
         };
 
-        let mut pkg_dep_graph =
-            PackageGraph::builder(&self.base.repo_root, root_package_json.clone())
-                .with_single_package_mode(opts.run_opts.single_package)
-                .with_package_discovery(
+        // if we are forcing the daemon, we don't want to fallback to local discovery
+        let (fallback, duration) = if let Some(true) = opts.run_opts.daemon {
+            (None, Duration::MAX)
```

Comment:
I think we need to error here if `daemon` is `None` instead of continuing as otherwise we're going to construct a `FallbackPackageDiscover::new(None, None, Duration::MAX)`. Maybe we should throw if we fail to connect to the daemon when `--daemon` is passed.

Reply:
It's a good point. I think I am going to make this really strict to the point where if you don't have a daemon connection in forced mode it crashes and make that requirement really clear.